src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage, injectIntl, FormattedNumber } from 'react-intl';
+import Loading from '../../Loading/Loading';
+
+const ReportsTrafficSources = function({
+  domainName,
+  dateFrom,
+  intl,
+  dependencies: { datahubClient },
+}) {
+  const [trafficSources, setTrafficSources] = useState({ loading: true });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const trafficSourcesData = await datahubClient.getTrafficSourcesByPeriod(
+        domainName,
+        dateFrom,
+      );
+      if (trafficSourcesData.success && trafficSourcesData.value) {
+        const total = trafficSourcesData.value.reduce(function(previous, item) {
+          return previous + item.quantity;
+        }, 0);
+        setTrafficSources({ loading: false, total: total, items: trafficSourcesData.value });
```

Comment:
I would wrap it in `value` or `trafficSources` object: 

```suggestion
        setTrafficSources({ loading: false, value: { total: total, items: trafficSourcesData.value }});
```

It allows to make it easy when try to show the value and when not.

Also, you could omit _loading_ property.